matscipy/cauchy_born.py

```diff
@@ -0,0 +1,742 @@
+        This means it should be applied to an unstrained version of the atoms structure and any sublattices
+        assigned adjacent to free surfaces cannot be trusted. This function updates self.lattice1mask and self.lattice2mask with the
+        measured sublattices.
+        Parameters
+        ----------
+        atoms : ASE atoms object
+            atoms structure with atoms on seperate sublattices that need assigning
+        A : 3x3 numpy array, floats
+            rotation matrix of the form [x^T,y^T,z^T], where x^T,y^T,z^T are normalised column vectors of the lab frame directions expressed
+            in terms of the crystal lattice directions.
+        '''
+
+        #currently, the idea for this is to apply a small strain to atoms
+        #measure the force difference, and set all the forces which are positive to sublattice
+        #1, and all those which are negative to sublattice 2. 
+
+        #generate U (right hand stretch tensor) to apply
+        U_voigt = np.array([1.001, 1.003, 1.002, 0.006, 0.002, 0.004])
+        U = np.zeros([3,3])
+        U[0,0] = U_voigt[0]
+        U[1,1] = U_voigt[1]
+        U[2,2] = U_voigt[2]
+        U[1,2],U[2,1] = U_voigt[3],U_voigt[3]
+        U[0,2],U[2,0] = U_voigt[4],U_voigt[4]
+        U[0,1],U[1,0] = U_voigt[5],U_voigt[5]
+        atoms_copy = atoms.copy()
+
+        #generate a copy of the atoms and stretch the cell, scaling atoms.
+        #get forces before and after and compare these
+        atoms_copy.set_calculator(self.calc)
+        f_before = atoms_copy.get_forces()
+        cell = atoms_copy.get_cell()
+        scaled_cell = U@cell
+        atoms_copy.set_cell(scaled_cell,scale_atoms=True)
+        f_after = atoms_copy.get_forces()
+
+        force_diff = f_after-f_before
+
+        #transform force_diff to lattice frame
+        force_diff_lattice = np.zeros_like(force_diff)
+        for i in range(len(atoms)):
+            force_diff_lattice[i,:] = A@force_diff[i,:]
+
+        #generate the lattice masks, checking to ensure that if by chance no shift occurs in one direction due to the applied strains 
+        #that the other directions are used instead and things don't break
+        lattice1mask = force_diff_lattice[:,0]>0
+        lattice2mask = force_diff_lattice[:,0]<0
+        if all(element == lattice1mask[0] for element in lattice1mask): #if they're all false or true, try the y component
+            lattice1mask = force_diff_lattice[:,1]>0
+            lattice2mask = force_diff_lattice[:,1]<0
+            if all(element == lattice1mask[0] for element in lattice1mask): #if they're all false, try the z component
+                lattice1mask = force_diff_lattice[:,2]>0
+                lattice2mask = force_diff_lattice[:,2]<0
+        
+
+        #set the masks for which atoms are in each of the sublattices
+        self.lattice1mask = lattice1mask
+        self.lattice2mask = lattice2mask     
+
```

Comment:
Style: only one blank line between methods in a class (two between functions in module). You could use a pep8 linter/formatter to fix automatically.